blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmContentProvider.java

```diff
@@ -0,0 +1,88 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import hudson.Extension;
+import io.jenkins.blueocean.commons.ServiceException;
+import io.jenkins.blueocean.rest.impl.pipeline.scm.ScmContent;
+import io.jenkins.blueocean.rest.impl.pipeline.scm.ScmContentProvider;
+import jenkins.scm.api.SCMFile;
+import jenkins.scm.api.SCMSource;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.StringUtils;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMSource;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+/**
+ * @author Vivek Pandey
+ */
+@Extension(ordinal = -100)
```

Comment:
Haha, Yes it is to allow others to provide their own version of GitHub content provider.